### DISCUSSION TITLE
How to Intercept auth() Handler in MSW for Mocking Server Session

### DISCUSSION THREAD

**Discussion by Rikard-Johansson97** [nextauthjs/next-auth#9576] [Help]
## How to Intercept `auth()` Handler in MSW for Mocking Server Session

Hello,

I'm currently working on a project where I'm using [Mock Service Worker (MSW)](https://mswjs.io/) to handle API mocking. I've set up my server and handlers as follows:

```javascript
// src/mocks/server.js
import { setupServer } from 'msw/node';
import { handlers } from './handlers';

// This configures a request mocking server with the given request handlers.
export const server = setupServer(...handlers);
```

```javascript
// src/mocks/handlers.js
import { rest } from 'msw';
import { mockUser } from '../preview';

export const handlers = [
  rest.get('/api/auth/session', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(mockUser));
  }),
];
```

Could anyone guide me on how to approach this problem? Specifically, I'm looking for a way to:

Intercept the auth() handler within MSW.
Mock the server session effectively after intercepting this handler.

**Comment by DiegoGonzalezCruz** [comment]
could you fix it? I'm trying to do the same

**Reply by Rikard-Johansson97** [reply]
I was able to mock the getSession and useSession with the MSW 
`GET: /api/auth/session`

But I think I need to understand a bit more how the auth() handlers is working to be able to mock them.

**Reply by NikkiMather** [reply]
I don't understand? In your original post it has `GET` for `/api/auth/session`. In your response you said you used `GET: /api/auth/session`, so how was you able to mock it using the exact same string?

**Comment by jhroemer** [comment]
Hi @Rikard-Johansson97 I'm looking into something similar currently. Did you find a good solution?
